### PR TITLE
[koa-generic-session] Types for koa.Context

### PR DESCRIPTION
**Pull request by hanakla**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
  - But got below errors (How can I fix it?)
    ```
    ./DefinitelyTyped/types/koa-generic-session/index.d.ts
    ERROR: 58:30  expect  TypeScript compile error: Cannot find name 'Generator'.

    ./DefinitelyTyped/types/koa-generic-session/koa-generic-session-tests.ts
    ERROR: 43:21  expect  TypeScript compile error: Property 'key' does not exist on type 'Session'.
    ````

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/koajs/generic-session#example
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/koa-generic-session/index.d.ts [hunk 1]
```diff
@@ -50,5 +51,13 @@ declare namespace koaSession {
     const MemoryStore: SessionStore;
 }
 
+declare module 'koa' {
+    interface Context {
+        session: koaSession.Session|null;
+        sessionSave: boolean|null;
+        regenerateSession(): Generator;
```
Comment (nsimmons) [inline types/koa-generic-session/index.d.ts]: Where is `Generator` defined? It either needs to be defined here or imported from somewhere.

File: types/koa-generic-session/koa-generic-session-tests.ts [hunk 2]
```diff
@@ -32,4 +32,16 @@ app.use(session({
...
 
+app.use((context: Koa.Context) => {
+    if (! context.session) {
+        return;
+    }
+
+    context.regenerateSession();
+    context.sessionSave = true;
+    context.session.cookie;
+    context.session.key = 'value';
```
Comment (nsimmons) [inline types/koa-generic-session/koa-generic-session-tests.ts]: You can't access the session keys like this when you define it with `[key: string]: any;`. 

`context.session['key'] = 'value';` should work.

**Review by nsimmons — CHANGES_REQUESTED** [review]

File: types/koa-generic-session/index.d.ts (hunk 1 above)
Comment (Ragg-) [inline types/koa-generic-session/index.d.ts]: `Generator` defined in `lib.es6.ts` from TypeScript starndard library.
This type definition already setting up to import `es6`  lib, but got error 🤔 https://github.com/Ragg-/DefinitelyTyped/blob/8117819ce704fe73c3ac9f0db5cb9dfbf2b70f01/types/koa-generic-session/tsconfig.json#L5

Comment (nsimmons) [inline types/koa-generic-session/index.d.ts]: Oh I see.... hmm not sure what's wrong.

Comment (Ragg-) [inline types/koa-generic-session/index.d.ts]: `Generator` is defined since TypeScript 2.3, but linter is using < 2.3.

**Comment by andy-ms** [comment]
Thanks! Issue was Microsoft/dtslint#43

**Pull request merged**